.config/semgrep.yaml

```diff
@@ -205,6 +205,17 @@ rules:
   message: "semver::VersionReq doesn't work for comparing prereleases. Compare with semver::Version instead. For context, see https://github.com/fedimint/fedimint/pull/4804"
   severity: ERROR
 
+- id: no-placeholder-in-expect
+  message: "Placeholders ('{...}') should not be used in `.expect()` calls, these messages are static and not interpolated."
+  severity: ERROR
+  languages: [rust]
+  patterns:
+    - pattern: |
+        $EXPR.expect($MSG)
+    - metavariable-regex:
+        metavariable: $MSG
+        regex: ".*\\{.*\\}.*"
+
```

Comment:
vote to use clippy instead 
https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args

Reply:
`Added in: 1.85.0` , huh.

Yeah, I think whatever works ATM, but clippy would be preferable.

Reply:
Fixed, I also used the opportunity to introduce workspace-inherited clippy rules. We might want to use that for `clippy::pedantic` too, which right now is at the top of each `lib.rs` file.